3-enrich/scala-common-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/registry/CurrencyConversionEnrichmentSpec.scala

```diff
@@ -0,0 +1,61 @@
+import com.snowplowanalytics.forex.{Forex, ForexConfig}
+
+import com.snowplowanalytics.snowplow.enrich.common.enrichments.registry._
+// This project
+//import object com.snowplowanalytics.snowplow.enrich.common.enrichments.registry.CurrencyConversionEnrichment
+/**
+ * Tests the convertCurrencies function
+ */
+
+class CurrencyConversionEnrichmentSpec extends Specification with DataTables { def is =
+
+  "This is a specification to test convertCurrencies"                  ^
+                                                                      p^
+    "Currency Conversion should work"                               ! e1^
+                                                                      end
+
+  val appIdInvalid = (TransactionAmounts("Open Exchange Rates error, message: Invalid App ID provided - please sign up at https://openexchangerates.org/signup, or contact support@openexchangerates.org. Thanks!","Open Exchange Rates error, message: Invalid App ID provided - please sign up at https://openexchangerates.org/signup, or contact support@openexchangerates.org. Thanks!","Open Exchange Rates error, message: Invalid App ID provided - please sign up at https://openexchangerates.org/signup, or contact support@openexchangerates.org. Thanks!"),"Open Exchange Rates error, message: Invalid App ID provided - please sign up at https://openexchangerates.org/signup, or contact support@openexchangerates.org. Thanks!")
+  def e1 =
+    "SPEC NAME"                    || "TRANSACTION PRICE"      | "API Key"                                  | "TRANSACTION AMOUNT"                       | "TRANSACTION ITEM CURRENCY"     | "TRANSACTION ITEM PRICE"                |"CONVERTED TUPLE"                                                          |
+    "First Success Test"           !! "GBP"                    ! ""                                         ! TransactionAmounts("12.99", "2.67", "0")   ! "GBP"                           ! "12.99"                                 !(TransactionAmounts("8.65","1.78","0.0"),"8.65").success                   |
+    "First Failure Test"           !! "RUP"                    ! ""                                         ! TransactionAmounts("14.99", "1.17", "0")   ! "GBP"                           ! "17.99"                                 ! "Exception Converting Currency :[Either.left.value on Right]".failureNel  |
+    "Second Failure Test"          !! "HUL"                    ! ""                                         ! TransactionAmounts("10.99", "0.7",  "0")   ! "GBP"                           ! "1.99"                                  ! "Exception Converting Currency :[Either.left.value on Right]".failureNel  |
+    "Third Failure Test"           !! "GBP"                    ! "89ab8d8dabdj9usd9d8siasjndsajA8Ada8"      ! TransactionAmounts("2.99",  "3.67", "0")   ! "GBP"                           ! "2.99"                                  ! appIdInvalid.success                                                      |
```

Comment:
Waiting for your comment for this part specifically !! (Test Case for wrong AppId)